packages/frontend/src/pages/dashboard/[guildId]/modmail/settings/index.tsx

```diff
@@ -0,0 +1,126 @@
+import type { APIChannel, APIDMChannel, APIGroupDMChannel } from 'discord-api-types/v10';
+import { useMemo } from 'react';
+import ConfigForm from '~/components/Config/ConfigForm';
+import ConfigOption from '~/components/Config/ConfigOption';
+import { ConfigOptionCollection } from '~/components/Config/ConfigOptionCollection';
+import ConfigPageFrame from '~/components/Config/ConfigPageFrame';
+import { TextArea } from '~/components/Config/TextArea';
+import type { DropdownGroupedOptions } from '~/components/Dropdown';
+import Dropdown from '~/components/Dropdown';
+import PageMeta from '~/components/PageMeta';
+import * as Text from '~/components/Text';
+import useGuildInfo from '~/hooks/useGuildInfo';
+import useModmailSettings from '~/hooks/useModmailSettings';
+
+const allowedChannelTypes = [
+	0, // GUILD_TEXT
+	15, // GUILD_FORUM
+];
+
+const withIcons = false;
+type WithIcons = typeof withIcons;
+type ApplicableChannel = Exclude<APIChannel, APIDMChannel | APIGroupDMChannel>;
+type GroupedChannels = (DropdownGroupedOptions<WithIcons> & { position: number })[];
+
+function addToCategoryIfExists(
+	acc: GroupedChannels,
+	channel: ApplicableChannel,
+	channels: ApplicableChannel[],
+): GroupedChannels {
+	if (!acc.some(({ id }) => id === (channel.parent_id ?? '0'))) {
+		const category = channels.find(({ id }) => id === channel.parent_id);
+
+		return [
+			...acc,
+			{
+				id: channel.parent_id ?? '0',
+				label: category?.name?.toUpperCase() ?? 'UNCATEGORIZED',
+				position: category?.position ?? -1,
+				options: [
+					{
+						value: channel.id,
+						label: channel.name ?? 'Unknown Channel',
+					},
+				],
+			},
+		];
+	}
+
+	acc
+		.find(({ id }) => id === (channel.parent_id ?? '0'))!
+		.options.push({
+			value: channel.id,
+			label: channel.name ?? 'Unknown Channel',
+		});
+	return acc;
+}
+
+function ModMailSettings() {
+	const { data: guildInfo, isLoading: isGuildDataLoading } = useGuildInfo();
+
+	const channelOptions = useMemo(() => {
+		const channels = (guildInfo?.channels as ApplicableChannel[]) ?? [];
+
+		return (
+			channels
+				// @ts-expect-error TS2339: docs: "may be missing for some channel objects received over gateway guild dispatches", not applicable to us
```

Comment:
I am not fond of using a ts-expect-error here *at all*. It should only be used for outright incorrect/incomplete types in an upstream package/a compiler bug. Try to find a way to cast - e.g. if `channel.position!` does not work it sounds like you're doing some poor type narrowing somewhere, since it should always exist on a guild channel, if I am not mistaken.